### PR TITLE
Fix self-update failuers due to /runner/externals mount

### LINKED ISSUE #252: Self update failed due to `/externals` mount point
Hi,

We ran into this issue - whenever it tries to handle a job, it detects it needs to auto update and then crashes at the end. After adding a `sleep 3600` at the end, we were able to catch it.

Crash logs:
```
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:25:24.297624393Z 2021-01-11 17:25:24Z: Listening for Jobs
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:21.656887962Z Runner update in progress, do not shutdown runner.
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:21.761093347Z Downloading 2.275.1 runner
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:28.167734819Z Waiting for current job finish running.
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:28.253782165Z Generate and execute update script.
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:28.337678733Z Runner will exit shortly for update, should back online within 10 seconds.
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:31.420450852Z renamed '/runner/_diag/SelfUpdate-20210111-172728.log' -> '/runner/_diag/SelfUpdate-20210111-172728.log.failed'
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:31.421289389Z Runner listener exited with error code 3
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:31.421339241Z Runner listener exit because of updating, re-launch runner in 5 seconds.
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.427363934Z Starting Runner listener with startup type: service
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.428878354Z Started listener process
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438268920Z events.js:187
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438294986Z       throw er; // Unhandled 'error' event
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438299706Z       ^
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438303203Z
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438306351Z Error: spawn /runner/bin/Runner.Listener ENOENT
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438310107Z     at Process.ChildProcess._handle.onexit (internal/child_process.js:264:19)
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438313798Z     at onErrorNT (internal/child_process.js:456:16)
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438317296Z     at processTicksAndRejections (internal/process/task_queues.js:80:21)
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438320791Z Emitted 'error' event on ChildProcess instance at:
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438324025Z     at Process.ChildProcess._handle.onexit (internal/child_process.js:270:12)
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438326635Z     at onErrorNT (internal/child_process.js:456:16)
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438329041Z     at processTicksAndRejections (internal/process/task_queues.js:80:21) {
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438341082Z   errno: 'ENOENT',
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438344220Z   code: 'ENOENT',
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438346528Z   syscall: 'spawn /runner/bin/Runner.Listener',
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438348864Z   path: '/runner/bin/Runner.Listener',
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438351150Z   spawnargs: [ 'run', '--startuptype', 'service' ]
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438353627Z }
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438355820Z + trap - TERM INT
actions-runner-worker-fsjpf-kv9j4 runner 2021-01-11T17:27:36.438358144Z + wait 75
```

Self update failed log:

```
removed '/runner/bin/runsvc.sh.bak'
removed '/runner/bin/RunnerService.js.bak'
removed directory '/runner/bin'
[2021-01-11 17:32:47-5904] move /runner/externals /runner/externals.2.274.2
mv: cannot move '/runner/externals' to '/runner/externals.2.274.2': Device or resource busy
[2021-01-11 17:32:47-5923] Can't move /runner/externals to /runner/externals.2.274.2
```

So it seems that due to the `/runner/externals` mount caused it to fail because it's an emptyDir mount so it can't be moved... Any ideas on how to fix?

Thanks!

### PR DESCRIPTION
**Pull request by mumoshu**
Please follow the steps described in https://github.com/summerwind/actions-runner-controller#developing to build the container images for testing.

Fixes #252

### CODE REVIEW

File: runner/Dockerfile [hunk 1]
```diff
@@ -72,14 +77,14 @@ RUN export ARCH=$(echo ${TARGETPLATFORM} | cut -d / -f2) \
...
   && chgrp runner /opt/hostedtoolcache \
   && chmod g+rwx /opt/hostedtoolcache
 
-COPY entrypoint.sh /runner
-COPY patched /runner/patched
+COPY entrypoint.sh /
+COPY patched $RUNNER_ASSETS_DIR/patched
 
-USER runner
+USER runner/runner
```
Comment (jolestar) [inline runner/Dockerfile:88]: unable to find user runner/runner: no matching entries in passwd file, this is correct?

Comment (mumoshu) [reply runner/Dockerfile:88]: Sorry this should just be `USER runner`

**Comment by jolestar** [comment]
I try use new runner image, got:

```text
Github endpoint URL https://github.com/
mv: cannot move '/runnertmp/bin' to '/runner/bin': Permission denied
mv: cannot move '/runnertmp/config.sh' to '/runner/config.sh': Permission denied
mv: cannot move '/runnertmp/env.sh' to '/runner/env.sh': Permission denied
mv: cannot move '/runnertmp/externalstmp' to '/runner/externalstmp': Permission denied
mv: cannot move '/runnertmp/patched' to '/runner/patched': Permission denied
mv: cannot move '/runnertmp/run.sh' to '/runner/run.sh': Permission denied
/entrypoint.sh: line 56: ./config.sh: No such file or directory
mv: cannot stat './externalstmp/*': No such file or directory
diff: bin/runsvc.sh: No such file or directory
diff: patched/runsvc.sh: No such file or directory
mv: cannot stat 'bin/runsvc.sh': No such file or directory
mv: cannot stat 'patched/runsvc.sh': No such file or directory
diff: bin/RunnerService.js: No such file or directory
diff: patched/RunnerService.js: No such file or directory
mv: cannot stat 'bin/RunnerService.js': No such file or directory
mv: cannot stat 'patched/RunnerService.js': No such file or directory
/entrypoint.sh: line 68: /runner/bin/runsvc.sh: No such file or directory
```

**Comment by mumoshu** [comment]
@jolestar Thanks. Ah - Probably you need `sudo chown -R runner:runner /runner` in the very beginning of your entrypoint.sh

https://github.com/summerwind/actions-runner-controller/blob/master/runner/entrypoint.sh

**Comment by jolestar** [comment]
@mumoshu 

Need add "mkdir ./externals" at here 

https://github.com/summerwind/actions-runner-controller/blob/524cf1b3793aeb8ea6280b2c1cd0a9cd55090173/runner/entrypoint.sh#L51

the runnertmp dir also need to chown -R runner:docker.

It is work for me now.

**Comment by mumoshu** [comment]
> Need add "mkdir ./externals" at here

Ahhh, good catch! That makes sense. Thanks a lot for sharing your insight 👍

**Comment by jolestar** [comment]
HorizontalRunnerAutoscaler can not get correct DESIRED value, I'm trying to figure out why. May be it is github action api's bug, too many action job is timeout, and all job's status is weird

**Comment by achedeuzot** [comment]
In my custom runner logs, when I check the failure log, this is what I see:

```
$ cat SelfUpdate-20210121-092250.log.failed
[..]
[2021-01-21 09:22:55-0335] move /runner/externals /runner/externals.2.275.1
mv: cannot move '/runner/externals' to '/runner/externals.2.275.1': Device or resource busy
[2021-01-21 09:22:55-0366] Can't move /runner/externals to /runner/externals.2.275.1
$ command terminated with exit code 137
```

I don't have any custom `entrypoint.sh`, I'm using the default one.

**Comment by mumoshu** [comment]
Let's go ahead! Thanks for all your help @e36-cicd, @jolestar, @achedeuzot, @hfuss, and everyone involved ❤️

**Pull request merged**